raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -58,7 +58,12 @@ library ChannelManagerLibrary {
 
         address[] storage existingChannels = self.nodeChannels[msg.sender];
```

Comment:
The following for loop iterates over self.nodeChannels[msg.sender] and calls deleteChannel. deleteChannel itself also iterates over self.nodeChannels[msg.sender] and modifies the array itself. This looks a bit dangerous.

This is just a guess, but I think the loop may fail to examine the last element (or the last N elements) of the existingChannels array, since deleteChannel copies the last element to the index i and then i will be incremented.
 (for some i for which (!contractExists(self, existingChannels[i])) is true.

Please ask me if my explanation hasn't been clear enough.